### ISSUE TITLE
Logout path not accessible during two-factor authentication when using CSRF protection

### ISSUE THREAD

**Issue by mdriessen** [scheb/2fa#28]
**Bundle version**: 5.1.1
**Symfony version**: 4.4.11

**Description**
I'm using CSRF protection on my logout URL. This is done by using csrf_token_generator: security.csrf.token_manager instead of true on the logout setting of the firewall. The logout url looks something like: https://mysite/logout?_csrf_token=9UF0Jk4bgAMuTBQcUGkKVNila9U1Vv5vrzC5-3WmQZc

When on the login page of the two-factor authentication, I can't get back to the login page. In the TwoFactorAccessDecider this part is not working with the GET variable, it only works on absolute path's.
```
// Let the logout route pass
$logoutPath = $this->makeRelativeToBaseUrl($this->logoutUrlGenerator->getLogoutPath(), $request);
if ($this->httpUtils->checkRequestPath($request, $logoutPath)) {
    return true;
}
```

**To Reproduce**
1. Enable csrf_token_generator on the logout URL.
2. Enable 2FA
3. Login to get to the 2FA login
4. Press back to return to the login screen
5. Access is denied and page refreshes

**Comment by scheb** [comment]
Confirmed, bug fix on the way. Travis build is somewhat stuck at the moment.

Btw, if you have an access control rule for the logout route, you shouldn't have this problem at all. Example here: https://github.com/scheb/2fa/blob/5.x/doc/installation.md#step-4-configure-the-firewall

**Comment by mdriessen** [comment]
Thank you for the quick reply and fix! The access control rule doesn't seem to work, I still get the 'User is in a two-factor authentication process.' exception. I believe the builtin Symfony AuthenticatedVoter votes DENIED because the user not not anonymous during 2FA.

**Comment by scheb** [comment]
Fix released as v5.1.2.

Access control shouldn't vote "denied" for an anonymous route. In Symfony any state (not authenticated, partially authenticated, fully authenticated) evaluates as `IS_AUTHENTICATED_ANONYMOUSLY`. There must be something odd in your application. But anyways, the bugfix should do the job for you.

**Comment by mdriessen** [comment]
Confirmed v5.1.2 is working for me. Thanks for the quick response!
